pkg/model/flamegraph.go

```diff
@@ -29,7 +29,7 @@ func NewFlameGraph(t *Tree, maxNodes int64) *querierv1.FlameGraph {
 
 	minVal := t.minValue(maxNodes)
 
-	stack := stackNodePool.Get().(*Stack[stackNode])
+	stack := getNodeStackPool().Get().(*Stack[stackNode])
```

Comment:
I dont get it. Does this create a fresh pool and fresh stack every time?

Reply:
![image](https://github.com/user-attachments/assets/d5fa7877-e8e1-464b-b503-ddc60cc8aad0)